### DISCUSSION TITLE
Log errors by jobs

### DISCUSSION THREAD

**Discussion by pnicosia80** [oqtane/oqtane.framework#4353] [Q&A] [answered]
Hello,
from my job I write errors in the event log of Oqtane using the method

_logManager.Log(site.SiteId, LogLevel.Error, this, LogFunction.Other, "message");

I absolutely need to do this, in addition to logging from the job, to expose information, warnings and errors on my dashboard. Unfortunately, in case of an error Oqtane sends a notification, but using getAlias method from the job without httpcontext causes an exception and prevents me from inserting it. 
In my opinion, in LogManager.cs, it would be enough to insert a check in the method below so that if the alias is null it does not send the notification in order to allow jobs to log errors as well.

![image](https://github.com/oqtane/oqtane.framework/assets/10197467/69c5fed4-3c43-43b0-8a2e-9dbd6696a8cc)

Thanks

**Comment by sbwalker** [comment]
@pnicosia80 in your scheduled job you could call TenantManager.SetAlias(tenantid, siteid) to ensure the Alias is set - which would avoid the exception in the SendNotification method.

**Reply by pnicosia80** [reply]
Even setting the alias [_tenantManager.SetAlias(site.TenantId, site.SiteId)], I have an error for null httpcontext. How can I solve this?

![image](https://github.com/oqtane/oqtane.framework/assets/10197467/8a2046c4-f9a9-415d-bf86-da1b2ac6ae1e)

**Reply by sbwalker** [reply]
@pnicosia80 yes,, the _accessor.HttpContext is problematic as well - this method will need to be modified to handle scenarios where there is no HttpContext. I will include this for the next release.

**Reply by pnicosia80** [reply]
Thank you for your availability, so I'll wait for the next release on git.

**Comment by sbwalker** [comment]
@pnicosia80 I do agree with your suggestion that this logging logic could be more defensive to prevent exceptions.

**Comment by sbwalker** [comment, marked as answer]
@pnicosia80 #4356 addresses your issue:

- defensive logic added to validate Alias is not null
- use Alias.Protocol rather than accessor.HttpContext.Request.Protocol to avoid HttpContext exceptions
- use GetUserRoles(roleName, siteId) to get list of host users rather than using GetUserRoles(siteId) which retrieves all UserRole records for the entire site (performance improvement)

**Reply by pnicosia80** [reply]
I have updated and compiled various fixes from git, but now at startup of oqtane and at regular intervals, it logs these exceptions

![image](https://github.com/oqtane/oqtane.framework/assets/10197467/bbe3eb25-77ad-4265-8a7e-710b0f4efdae)

**Reply by sbwalker** [reply]
@pnicosia80 the Dev branch is still under active development for the features to be included in the 5.2.0 release. There may still be additional changes to interfaces, database tables, etc... (ie. the Dev branch should not be deployed to any production environment ). Thank you for identifying this issue... I will make sure it gets resolved before the 5.2.0 release.

**Comment by pnicosia80** [comment]
@sbwalker  Thanks! I have resolved the issue of error logging from the job.
